### PR TITLE
Fixes MOVSD

### PR DESCRIPTION
**Pull request by sushant94**

### CODE REVIEW

File: libr/anal/p/anal_x86_cs.c [hunk 1]
```diff
@@ -450,8 +488,24 @@ Sets the byte in the operand to 1 if the Sign Flag is not equal
...
 	case X86_INS_MOVSD:
+		// Handle "Move Scalar Double-Precision Floating-Point Value"
+		if (is_xmm_reg(INSOP(0)) || is_xmm_reg(INSOP(1))) {
+			const char *src = getarg(&gop, 1, 0, NULL);
+			const char *dst = getarg(&gop, 0, 1, NULL);
+			esilprintf (op, "%s,%s", src, dst);
+			free(src);
+			free(dst);
+			break;
+		}
```
Comment (zamaudio) [inline libr/anal/p/anal_x86_cs.c]: is the other case of movsd currently unhandled?

Comment (sushant94) [inline libr/anal/p/anal_x86_cs.c]: It is handled in the fall through of the switch case.

File: libr/anal/p/anal_x86_cs.c [hunk 2]
```diff
@@ -29,6 +29,44 @@ struct Getarg {
 	int bits;
 };
 
+int is_xmm_reg(cs_x86_op op) {
```
Comment (radare) [inline libr/anal/p/anal_x86_cs.c]: make this static, doesnt needs to be public

Comment (radare) [inline libr/anal/p/anal_x86_cs.c]: make it return bool

File: libr/anal/p/anal_x86_cs.c [hunk 3]
```diff
@@ -29,6 +29,44 @@ struct Getarg {
...
+	switch (op.reg) {
```
Comment (radare) [inline libr/anal/p/anal_x86_cs.c]: wrong indent

File: libr/anal/p/anal_x86_cs.c [hunk 4]
```diff
@@ -29,6 +29,44 @@ struct Getarg {
...
+		case X86_REG_XMM22:
+		case X86_REG_XMM23:
+		case X86_REG_XMM24:
+		case X86_REG_XMM25:
+		case X86_REG_XMM26:
+		case X86_REG_XMM27:
+		case X86_REG_XMM28:
+		case X86_REG_XMM29:
+		case X86_REG_XMM30:
+		case X86_REG_XMM31: return 1;
```
Comment (radare) [inline libr/anal/p/anal_x86_cs.c]: return true/false

File: libr/anal/p/anal_x86_cs.c [hunk 5]
```diff
@@ -450,8 +488,24 @@ Sets the byte in the operand to 1 if the Sign Flag is not equal
...
+	// has the same pneumonic for two different opcodes!). We can decide which
+	// of the two it is baed on the operands.
+	// For more information, see:
+	// http://x86.renejeschke.de/html/file_module_x86_id_203.html
+	//               (vs)
+	// http://x86.renejeschke.de/html/file_module_x86_id_204.html
 	case X86_INS_MOVSD:
+		// Handle "Move Scalar Double-Precision Floating-Point Value"
+		if (is_xmm_reg(INSOP(0)) || is_xmm_reg(INSOP(1))) {
+			const char *src = getarg(&gop, 1, 0, NULL);
```
Comment (radare) [inline libr/anal/p/anal_x86_cs.c]: add spaces before parenthesis

File: libr/anal/p/anal_x86_cs.c [hunk 6]
```diff
@@ -450,8 +488,24 @@ Sets the byte in the operand to 1 if the Sign Flag is not equal
...
+			const char *dst = getarg(&gop, 0, 1, NULL);
+			esilprintf (op, "%s,%s", src, dst);
```
Comment (radare) [inline libr/anal/p/anal_x86_cs.c]: check if src && dst are != NULL before using them

File: libr/anal/p/anal_x86_cs.c [hunk 7]
```diff
@@ -450,8 +488,24 @@ Sets the byte in the operand to 1 if the Sign Flag is not equal
 			r_strbuf_appendf (&op->esil, "rsi,[8],rax,=,df,?{,8,rsi,-=,},df,!,?{,8,rsi,+=,}");
 		break;
 	// string mov
-	case X86_INS_MOVSB:
+	// PS: MOVSD can correspond to one of the two instruction (yes, intel x86
+	// has the same pneumonic for two different opcodes!). We can decide which
+	// of the two it is baed on the operands.
```
Comment (ret2libc) [inline libr/anal/p/anal_x86_cs.c]: typo ;)

**Pull request merged**